From 68b818fcf735fd521db332733f9f61324afd8093 Mon Sep 17 00:00:00 2001
From: Lawrence Mok <lawrence.mok@amlogic.com>
Date: Thu, 30 Oct 2014 14:51:58 -0700
Subject: [PATCH 5005/5965] change sd debugfs permissions

For CTS.
in /sys/kernel/debug/{sd/sd,sdio/sdio,emmc/emmc}/, files:

params - read only, so changed to 444
sdio_dbg, sdhc_dbg - for debugging, so changed to 644
sdio_regs, sdhc_regs - read only, so changed to 444

Change-Id: Ic169cf61434c005a5de249f656d34d77068cdc03
---
 drivers/amlogic/mmc/amlsd.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/amlogic/mmc/amlsd.c b/drivers/amlogic/mmc/amlsd.c
index c2d11d163629..5389e01bb4a6 100755
--- a/drivers/amlogic/mmc/amlsd.c
+++ b/drivers/amlogic/mmc/amlsd.c
@@ -588,14 +588,14 @@ void aml_sdhc_init_debugfs(struct mmc_host *mmc)
 		goto err;
 	}
 
-	debugfs_create_x32("sdhc_dbg", S_IRWXUGO, root, (u32 *)&sdhc_debug);
+	debugfs_create_x32("sdhc_dbg", S_IRUGO | S_IWUSR, root, (u32 *)&sdhc_debug);
 
-	node = debugfs_create_file("sdhc_regs", S_IRWXUGO, root, mmc,
+	node = debugfs_create_file("sdhc_regs", S_IRUGO, root, mmc,
                 &aml_sdhc_regs_fops);
 	if (IS_ERR(node))
 		return;
 
-	node = debugfs_create_file("params", S_IRWXUGO, root, mmc,
+	node = debugfs_create_file("params", S_IRUGO, root, mmc,
                 &amlsd_param_fops);
 	if (IS_ERR(node))
 		return;
@@ -627,14 +627,14 @@ void aml_sdio_init_debugfs(struct mmc_host *mmc)
 		goto err;
 	}
 
-	debugfs_create_x32("sdio_dbg", S_IRWXUGO, root, (u32 *)&sdio_debug);
+	debugfs_create_x32("sdio_dbg", S_IRUGO | S_IWUSR, root, (u32 *)&sdio_debug);
 
-	node = debugfs_create_file("sdio_regs", S_IRWXUGO, root, mmc,
+	node = debugfs_create_file("sdio_regs", S_IRUGO, root, mmc,
                 &aml_sdio_regs_fops);
 	if (IS_ERR(node))
 		return;
 
-	node = debugfs_create_file("params", S_IRWXUGO, root, mmc,
+	node = debugfs_create_file("params", S_IRUGO, root, mmc,
                 &amlsd_param_fops);
 	if (IS_ERR(node))
 		return;
-- 
2.19.0

